runtime/registry.go

```diff
@@ -79,11 +72,148 @@ func (r *Runtime) CreateInstance(ctx context.Context, inst *drivers.Instance) er
 	return nil
 }
 
-func (r *Runtime) DeleteInstance(ctx context.Context, instanceID string) error {
-	err := r.Registry().DeleteInstance(ctx, instanceID)
+func (r *Runtime) DeleteInstance(ctx context.Context, instanceID string, dropDB bool) error {
+	inst, err := r.Registry().FindInstance(ctx, instanceID)
 	if err != nil {
 		return err
 	}
 
-	return nil
+	svc, err1 := r.catalogCache.get(ctx, r, instanceID)
+	var dropErrors []error
+	if dropDB {
+		// drop all ingested data
+		// should we insert complete data in one schema so it will be easy to drop the schema ?
+		// NOTE : Database file does not free space when tables are dropped issue #1099 in duckdb
+		entries := svc.Catalog.FindEntries(ctx, inst.ID, drivers.ObjectTypeUnspecified)
+		for _, entry := range entries {
+			if entry.Type == drivers.ObjectTypeTable {
+				table := entry.GetTable()
+				if table.Managed {
+					dropErrors = append(dropErrors, svc.Olap.Exec(ctx, &drivers.Statement{Query: fmt.Sprintf("DROP TABLE %s", table.Name)}))
+				}
+			}
+		}
+		// drop rill schema which deletes embedded catalog as well
+		dropErrors = append(dropErrors, svc.Olap.Exec(ctx, &drivers.Statement{Query: "DROP schema rill CASCADE"}))
+	}
```

Comment:
The goal of `dropDB` is to drop the actual `.db` file for DuckDB. For other OLAPs, it might be a `drop database ...` or something like that. So it should be a function on the `OLAPStore` interface. For DuckDB, it should figure out the DB filename (by parsing the DSN) and drop the file. For Druid, you can have it return an error for now (something like "dropping database not supported").

Reply:
Btw, it should be idempotent in case the DB was already dropped (in case of retries)

Reply:
If the DB has already been dropped, the API can't be retried since without db file we can't open db handlers (if already closed).